core/src/main/scala/org/apache/spark/rapids/tool/benchmarks/Benchmark.scala

```diff
@@ -121,16 +120,16 @@ class Benchmark(
    */
   def measure(num: Long, overrideNumIters: Int)(f: ToolsTimer => Unit): Result = {
     System.gc()  // ensures garbage from previous cases don't impact this one
-    val warmupDeadline = warmupTime.fromNow
-    while (!warmupDeadline.isOverdue) {
+    var wi = 0
+    while (wi < warmUpIterations) {
```

Comment:
A for-loop is better since it is defined steps

Reply:
Updated